solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -2192,6 +2191,9 @@ public SolrCore getCore(String name) {
    * @see SolrCore#close()
    */
   public SolrCore getCore(String name, UUID id) {
+    if (name == null) {
+      return null;
+    }
```

Comment:
I want to say this caused issues. Let me take a second look.

Reply:
Line 2211 - getCoreInitFailures can't have null keys and so looking up name is null caused issue. This just shortcircuits since null key would never match anyway. The Map.of is more strict than the existing ImmutableMap around null keys. This fixed the issue and returns the same null we would have gotten anyway (desc is null on line 2207 and then line 2221 returns null)